ford/utils.py

```diff
@@ -318,3 +332,135 @@ def sub_macros(string):
+            extDict['extends'] = intObj.extends
+        for attrib in attribs:
+            if hasattr(intObj, attrib):
+                if type(getattr(intObj, attrib)) == str:
+                    extDict[attrib] = getattr(intObj, attrib)
+                elif type(getattr(intObj, attrib)) == list:
+                    extDict[attrib] = []
+                    for item in getattr(intObj, attrib):
+                        extItem = obj2dict(item)
+                        extDict[attrib].append(extItem)
+                elif type(getattr(intObj, attrib)) == dict:
+                    extDict[attrib] = {}
+                    for key, val in getattr(intObj, attrib).items():
+                        extItem = obj2dict(val)
+                        extDict[attrib][key] = extItem
+        return extDict
+
+    def modules_from_local(url):
+        '''
+        Get module information from an external project but on the
+        local file system.
+        Uses the io module to work in both, Python 2 and 3.
+        '''
+        from io import open
+        with open(os.path.join(url, 'modules.json'),
+                  mode = 'r',
+                  encoding = 'utf-8' ) as extfile:
+            extModules = json.loads(extfile.read())
+        return extModules
+
+    def dict2obj(extDict, url, parent=None):
+        '''
+        Converts a dictionary to an object.
+        '''
+        if extDict['obj'].lower() == 'module':
+            extObj = ford.sourceform.ExtModule()
+            project.extModules.append(extObj)
+        elif extDict['obj'].lower() == 'proc':
+            if extDict['proctype'].lower() == 'function':
+                extObj = ford.sourceform.ExtFunction()
+            elif extDict['proctype'].lower() == 'subroutine':
+                extObj = ford.sourceform.ExtSubroutine()
+            elif extDict['proctype'].lower() == 'interface':
+                extObj = ford.sourceform.ExtInterface()
+            project.extProcedures.append(extObj)
+            extObj.proctype = extDict['proctype']
+        elif extDict['obj'].lower() == 'interface':
+            extObj = ford.sourceform.ExtInterface()
+            project.extInterfaces.append(extObj)
+            extObj.proctype = extDict['proctype']
+        elif extDict['obj'].lower() == 'type':
+            extObj = ford.sourceform.ExtType()
+            project.extTypes.append(extObj)
+            extObj.extends = extDict['extends']
+        elif extDict['obj'].lower() == 'variable':
+            extObj = ford.sourceform.ExtVariable()
+            project.extVariables.append(extObj)
+        extObj.name = extDict['name']
+        if extDict['extURL']:
+            extDict['extURL'] = '/' + extDict['extURL'].split('/', 1)[-1]
+            extObj.extURL = url + extDict['extURL']
+        else:
+            extObj.extURL = extDict['extURL']
+        extObj.obj = extDict['obj']
+        extObj.parent = parent
+        for key in attribs:
+            if key in extDict:
```

Comment:
Here, I'd be tempted to do:
```suggestion
            if key not in extDict:
                continue
```
Then the rest of the loop can be indented less

Reply:
That's also fine by me.